src/torcontrol.cpp

```diff
@@ -369,10 +416,7 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
         // Now that we know Tor is running setup the proxy for onion addresses
         // if -onion isn't set to something else.
         if (gArgs.GetArg("-onion", "") == "") {
```

Comment:
This line did cause me a problem and this is why I created https://github.com/bitcoin/bitcoin/pull/22830

In a nutshell, you could have the  onion proxy set with the following cmdline:
-proxy=192.168.1.2:9050 -torcontrol=127.0.0.1:9050

I believe that check the presence of the -onion switch is not the right check.

The code should instead call GetProxy() and only enter the conditional block if function returns false because the onion proxy is not currently valid.

Reply:
I think the current behaviour is correct.